src/libraries/Native/Unix/System.Security.Cryptography.Native.Apple/pal_x509.c

```diff
@@ -257,6 +257,21 @@ int32_t AppleCryptoNative_X509CopyPrivateKeyFromIdentity(SecIdentityRef identity
     return SecIdentityCopyPrivateKey(identity, pPrivateKeyOut);
 }
 
+int32_t AppleCryptoNative_X509GetRawData(SecCertificateRef cert, CFDataRef* ppDataOut, int32_t* pOSStatus)
+{
+    if (ppDataOut != NULL)
+        *ppDataOut = NULL;
+    if (pOSStatus != NULL)
+        *pOSStatus = noErr;
```

Comment:
Put these inside the subsequent if block?

Reply:
I left the error checking from the previous code. I can change it if desired.

Right now I'm still pondering about the general approach which is why I kept this as draft for now (submitted to test CI on your machines).